src/main/scala/uk/gov/hmrc/ct/accounts/AC12.scala

```diff
@@ -53,6 +70,65 @@ case class AC12(value: Option[Int]) extends CtBoxIdentifier(name = "Current Turn
     boxRetriever.ac3().value.plusMonths(12) <= boxRetriever.ac4().value
   }
 
+  private def daysBetweenDates(start: LocalDate, end: LocalDate): Int = {
+    Days.daysBetween(start, end).getDays + 1
+  }
+
+  private def isLeapYear(year: Int): Boolean = {
+    val beginningOfYear = new LocalDate(year, 1, 1)
+    (daysBetweenDates(beginningOfYear, beginningOfYear.plusYears(1)) - 1) == 366
+  }
+
+  private def getDaysInYear(boxRetriever: AccountsBoxRetriever): Int = {
+    val poaStartDate = boxRetriever.ac3().value
+    val poaEndDate = boxRetriever.ac4().value
+    val startYearIsLeap = isLeapYear(poaStartDate.getYear)
+    val endYearIsLeap = isLeapYear(poaEndDate.getYear)
+
+    val currentYearLeapDay =
+      if (startYearIsLeap)
+        new LocalDate(poaStartDate.getYear, 2, 29)
+      else // <<HACK, HACK, HACK>>
```

Comment:
Those "junk" dates aren't even used because of the way boolean expressions are read.